Synergy/Controllers/RandomCodeController.cs

```diff
@@ -0,0 +1,76 @@
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using Synergy.Authorization;
+using Synergy.Models;
+using Synergy.Services;
+using Permissions = Synergy.Authorization.Permissions;
+
+namespace Synergy.Controllers;
+
+[ApiController]
+[Route("/api/[controller]")]
+public class RandomCodeController : ControllerBase
+{
+    private readonly RandomCodeService _randomCodeService;
+
+    public RandomCodeController(RandomCodeService randomCodeService)
+    {
+        _randomCodeService = randomCodeService;
+    }
+
+    [HttpPost("generate-code")]
+    [Authorize]
+    [AuthorizePermission(Permissions.CanRegisterTeam)]
+    public async Task<IActionResult> GenerateCode([FromBody] TeamId body)
+    {
+        var teamId = body.Id;
+        try
+        {
+            var randomCode = await _randomCodeService.GenerateRandomCodeInvite(teamId);
+            return Ok(randomCode);
+        }
+        catch (Exception ex)
```

Comment:
You don't need to add in this catch (Exception ex), you can instead use an ErrorHandlingMiddleware. Something like this: https://www.milanjovanovic.tech/blog/global-error-handling-in-aspnetcore-8